search/utils.py

```diff
@@ -0,0 +1,366 @@
+
+    This is because the actual search result has nested types, which
+    would be cumbersome to access from arbitrary calling functions.
+    """
+    def __init__(self, search):
+        """Get raw search result from Elasticsearch."""
+        self._search = search
+
+    def count(self):
+        """Total records matching the query."""
+        return self._search.count()
+
+    def page_count(self):
+        """Total number of result pages."""
+        count = self._search.count() / PAGE_LENGTH
+        if self._search.count() % PAGE_LENGTH > 0:
+            count += 1
+        return int(count)
+
+    def get_page(self, page):
+        """Return paginated results."""
+        start = (page - 1) * PAGE_LENGTH
+        end = start + PAGE_LENGTH
+        return self._search[start:end].execute().hits
+
+    def all(self):
+        """Return all results in a generator."""
+        for i in range(self.page_count()):
+            for hit in self.get_page(i+1):
+                yield hit
+
+    def __getitem__(self, i):
+        """Return result by index."""
+        return self._search[i:i+1].execute().hits[0]
+
+
+def create_mapping():
+    """
+    Create the Elasticsearch mapping.
+
+    Notes on the "index" values:
+        analyzed: normal, will be split on whitespace for indexing
+            (partial matches can be made during queries)
+        not_analyzed: no split; matches must be exact
+        no: not indexed at all, but will be returned in search results
+    """
+
+    # Create the index if it doesn't exist.
+    if not CONN.indices.exists(INDEX_NAME):
+        CONN.indices.create(INDEX_NAME)
+    # Delete the mapping if an older version exists.
+    if CONN.indices.exists_type(index=INDEX_NAME, doc_type=DOC_TYPE):
+        CONN.indices.delete_mapping(index=INDEX_NAME, doc_type=DOC_TYPE)
+
+    mapping = Mapping(DOC_TYPE)
+
+    mapping.field("course", "string", index="not_analyzed")
+    mapping.field("description_path", "string", index="no")
+    mapping.field("description", "string", index="analyzed")
+    mapping.field("preview_url", "string", index="no")
+    mapping.field("repository", "string", index="not_analyzed")
+    mapping.field("resource_type", "string", index="analyzed")
+    mapping.field("content_xml", "string", index="no")
+    mapping.field("content_stripped", "string", index="analyzed")
+    mapping.field("run", "string", index="not_analyzed")
+    mapping.field("titlesort", "string", index="no")
+    mapping.field("title", "string", index="analyzed")
+
+    mapping.field("xa_avg_grade", "float")
+    mapping.field("xa_histogram_grade", "float")
+    mapping.field("xa_nr_attempts", "integer")
+    mapping.field("xa_nr_views", "integer")
+
+    mapping.save(INDEX_NAME)
+
+    # If we've done this, we always want to re-index all existing
+    # LearningResource instances. This function will probably only
+    # ever be called by migrations.
+    index_resources(LearningResource.objects.all())
```

Comment:
This is possible, but any concerns regarding using the real ORM are only possible during schema migrations. This is just a data migration.

There are other potential problems with using the fake (migration) models. In the event of any changes to the mapping or the model itself, the schema migrations would have run prior to the data migration, so everything will always be in the correct state.

I lean towards leaving it as is.